R/zzz.R

```diff
@@ -1,5 +1,27 @@
 .lockbox_env <- new.env()
 
+.get_safely_from_envir <- function(target, default = NULL) {
+  res <- tryCatch(
+    get(target, envir = .lockbox_env)
+  , error = function(e) {default})
+}
+
+set_session_id <- function() {
+  if (is.null(.get_safely_from_envir("session_id"))) {
+    if (requireNamespace("uuid", quietly = TRUE)) {
+      assign("session_id", uuid::UUIDgenerate(), envir = .lockbox_env)
+    } else {
+      assign("session_id", digest::digest(as.integer(Sys.time())), envir = .lockbox_env)
+    }
+  }
+  
+  # We don't want to clutter up ~/.R/ with tons of symlinked dirs, lets try to do some housekeeping
+  reg.finalizer(.lockbox_env, function(env) {
```

Comment:
I like this solution.

Maybe use the helpers here to avoid problems if the paths change:

```r
# Note the paths should already be normalized here.
lapply(c(lockbox_transient_staging_dir(), lockbox_transient_dir(), lockbox_download_dir()), 
  function(path) {  try(unlink(x, recursive = TRUE, force = TRUE), silent = TRUE) })
```

I would even define a helper `lockbox_session_dirs` in [lockbox.R](https://github.com/robertzk/lockbox/blob/cf10b065e770e1cc5066b730985f40d494d3bcc7/R/lockbox.R#L145) for the above 3-vector.